From cc5f362cbf927d5b7f902c3a582f11ceb3f789ad Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 3 Dec 2020 11:16:47 +0100
Subject: [PATCH] Fix check for sed --in-place with busybox mktemp

Fix build on Alpine Linux which uses busybox, and busybox mktemp
requires 6 trailing Xes.

Fixes commit 4313a5b4c51e (Detect if sed supports --in-place)

Signed-off-by: Natanael Copa <ncopa@alpinelinux.org>
Closes #11274
---
 config/always-sed.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/always-sed.m4 b/config/always-sed.m4
index 19633e118ae..3d7ae285ba1 100644
--- a/config/always-sed.m4
+++ b/config/always-sed.m4
@@ -4,7 +4,7 @@ dnl #
 AC_DEFUN([ZFS_AC_CONFIG_ALWAYS_SED], [
 	AC_REQUIRE([AC_PROG_SED])dnl
 	AC_CACHE_CHECK([for sed --in-place], [ac_cv_inplace], [
-		tmpfile=$(mktemp conftest.XXX)
+		tmpfile=$(mktemp conftest.XXXXXX)
 		echo foo >$tmpfile
 		AS_IF([$SED --in-place 's#foo#bar#' $tmpfile 2>/dev/null],
 		      [ac_cv_inplace="--in-place"],
